rpc/js/tsconfig.json

```diff
@@ -1,7 +1,7 @@
 {
   "extends": "@viamrobotics/typescript-config/base.json",
   "compilerOptions": {
-    "target": "ES5",
+    "target": "esnext",
     "useDefineForClassFields": true,
     "importHelpers": true,
```

Comment:
Some suggested TSConfig changes:

- Remove `useDefineForClassFields` and `importHelpers` as unnecessary
- Add `"noEmit": true`
    - Can then remove `--no-emit` from `scripts.typecheck` in `package.json`
- Remove `"verbatimModuleSyntax": false` to use our base default of `true`
    - This will catch all the `import type { ... };` problems, a few of which I pointed out
- Add `"noEmit": false` to `tsconfig.build.json` to keep the `.d.ts` file generation working

Reply:
didn't seem to catch `import { Message } from '@bufbuild/protobuf';` btw

Reply:
Yeah, I've seen it miss some stuff in the app, too. As long as we get the bulk of them that's good.

I think `vite` throws some build warnings about unused imports, too (haven't looking into how to promote those to errors, yet) and I suspect newer versions of `@typescript-eslint` might do a better job of detecting this, too, so hopefully we can get full coverage once this is over in the TS SDK repo